mne/io/ctf/ctf.py

```diff
@@ -149,13 +150,13 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
         with open(self._filenames[fi], 'rb') as fid:
             for bi in range(len(r_lims)):
                 samp_offset = (bi + trial_start_idx) * si['res4_nsamp']
-                n_read = min(si['n_samp'] - samp_offset, si['block_size'])
+                n_read = min(si['n_samp_tot'] - samp_offset, si['block_size'])
```

Comment:
As I said, our testing data has equal values for these. We should probably add dataset with different values. I wonder if @dengemann or someone else has nice small files.